src/include/compat.h

```diff
@@ -13,6 +13,7 @@
 #define CEPH_COMPAT_H
 
 #include "acconfig.h"
+#include "fcntl.h"
```

Comment:
please move this `#include` to the .cc file.

Reply:
@tchaikov 
It is already there, so I guess that there are reason why I put it there first place
But I'll remove, and see what happens in FreeBSD/Jenkins.

Reply:
@tchaikov 
Perhaps you have a suggestion, since this is an error on the Linux/GCC side of things.
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/include/compat.h:169:34: error: 'off_t' has not been declared
 int ceph_posix_fallocate(int fd, off_t offset, off_t len);
                                  ^~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/include/compat.h:169:48: error: 'off_t' has not been declared
 int ceph_posix_fallocate(int fd, off_t offset, off_t len);
                                                ^~~~~
src/CMakeFiles/crush_objs.dir/build.make:230: recipe for target 'src/CMakeFiles/crush_objs.dir/crush/CrushLocation.cc.o' failed
```

Reply:
please `#include <sys/types.h>` for `off_t`, see http://pubs.opengroup.org/onlinepubs/009696699/basedefs/sys/types.h.html .